tests/integration/targets/ndo_virtual_port_channel_interface/tasks/main.yml

```diff
@@ -0,0 +1,401 @@
+          - nm_update_virtual_port_channel_interface_descriptions.current.interfaceDescriptions.4.nodeID == "104"
+          - nm_update_virtual_port_channel_interface_descriptions.current.interfaceDescriptions.4.interfaceID == "1/2"
+          - nm_update_virtual_port_channel_interface_descriptions.current.interfaceDescriptions.4.description == "new second Ansible interface test for second node"
+          - nm_delete_virtual_port_channel_interface_descriptions is changed
+          - nm_delete_virtual_port_channel_interface_member is changed
+          - nm_delete_virtual_port_channel_interface_member.current.node1Details.memberInterfaces == "1/1-2"
+          - nm_delete_virtual_port_channel_interface_member.current.interfaceDescriptions | length == 2
+          - nm_delete_virtual_port_channel_interface_member.current.interfaceDescriptions.0.nodeID == "103"
+          - nm_delete_virtual_port_channel_interface_member.current.interfaceDescriptions.0.interfaceID == "1/1"
+          - nm_delete_virtual_port_channel_interface_member.current.interfaceDescriptions.0.description == "new first Ansible interface test for first node"
+          - nm_delete_virtual_port_channel_interface_member.current.interfaceDescriptions.1.nodeID == "103"
+          - nm_delete_virtual_port_channel_interface_member.current.interfaceDescriptions.1.interfaceID == "1/2"
+          - nm_delete_virtual_port_channel_interface_member.current.interfaceDescriptions.1.description == "new second Ansible interface test for first node"
+
+    # QUERY
+
+    - name: Create another virtual port channel interface
+      cisco.mso.ndo_virtual_port_channel_interface: &create_virtual_port_channel_interface_2
+        <<: *mso_info
+        template: ansible_fabric_resource_template
+        virtual_port_channel_interface: ansible_virtual_port_channel_interface_2
+        node_1: 101
+        node_2: 102
+        interfaces_1: 1/1
+        interfaces_2: 1/1
+        interface_policy_group: 
+          name: Gaspard_Interface_setting_test
+          template: Gaspard_FP_3.2_test
+        state: present
+
+    - name: Query a virtual port channel interface with template_name
+      cisco.mso.ndo_virtual_port_channel_interface:
+        <<: *create_virtual_port_channel_interface_2
+        state: query
+      register: query_one
+
+    - name: Query all virtual port channel interfaces in the template
+      cisco.mso.ndo_virtual_port_channel_interface:
+        <<: *mso_info
+        template: ansible_fabric_resource_template
+        state: query
+      register: query_all
+    
+    - name: Assert virtual port channel interface query tasks
+      assert:
+        that:
+          - query_one is not changed
+          - query_one.current.name == "ansible_virtual_port_channel_interface_2"
+          - query_all is not changed
+          - query_all.current | length == 2
```

Comment:
Should you also check that the 2 interfaces are the expected 2 returned? Maybe check the names of both are correct.

Reply:
why not, I will add a few assessment for checking if they are the correct ones